default.nix

```diff
@@ -0,0 +1,6 @@
+(import (let lock = builtins.fromJSON (builtins.readFile ./flake.lock);
+in fetchTarball {
+  url =
+    "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
```

Comment:
can you please elaborate on what URL that is, who maintains that, and how can we be sure this is safe to use?

Reply:
Sure thing. That's the URL for the flake-compat library which allows projects using the newer Flakes mechanism (i.e., what I'm adding with this PR) [to be used with derivations that don't use Flakes](https://nixos.wiki/wiki/Flakes#Using_flakes_project_from_a_legacy_Nix). It's maintained by Eelco Dolstra who wrote [the thesis that gave birth to the Nix package manager](https://edolstra.github.io/pubs/phd-thesis.pdf). The [source code for the library](https://github.com/edolstra/flake-compat/blob/master/default.nix) is somewhat small (200 lines), so it's feasible to review it quite quickly for anything malicious.